src/toxcore/dht/codec.rs

```diff
@@ -88,7 +88,10 @@ impl Decoder for DhtCodec {
 
         match DhtPacket::from_bytes(buf) {
             IResult::Incomplete(needed) => Err(DecodeError::IncompletePacket { len, needed }.into()),
-            IResult::Error(error) => Err(DecodeError::DeserializeError { error }.into()),
+            IResult::Error(error) => {
+                let error_kind = error.into_error_kind();
```

Comment:
Although we did not change the version of nom (3.2.1), it changed the enum of IResult.

@kurnevsky do you have any reason why that happened?

Reply:
You sure that you need this change? It seems `into_error_kind` is [no-op](https://docs.rs/nom/3.2.1/src/nom/util.rs.html#706-708) - it takes `ErrorKind` and returns it.

Reply:
It is a no-op for ErrorKind, whereas IResult::Error contains [Err\<T\>](https://github.com/Geal/nom/blob/1a70c8fa3cbb710c069395e1ee10019a4f8d3709/src/internal.rs#L133)

Reply:
Which is synonym to `ErrorKind`: https://docs.rs/nom/3.2.1/nom/simple_errors/type.Err.html

Reply:
It was changed to enum only in version 4.

Reply:
Oh... I got a strange error that `error` is a type of `Err<&[u8]>`, while `ErrorKind` was expected.

I reverted the code and it worked for me. T_T